### ISSUE TITLE
[VersionControl] Error fetching from Git server

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7104]
Fetching sometimes gives an error or without any error never ends.

```
| at MonoDevelop.VersionControl.Git.GitConfigurationDialog.OnButtonFetchClicked (System.Object sender, System.EventArgs e) [0x000af] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitConfigurationDialog.cs:375<br>  
at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] 
in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1023<br>  
at MonoDevelop.Ide.DispatchService+GtkSynchronizationContext+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00016] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77<br><br>----------<br><br>  
ERROR [2019-02-11 09:24:31Z]: An unhandled exception has occurred. 
Terminating MonoDevelop? False<br>System.NullReferenceException: Object reference not set to an instance of an object<br>  
at MonoDevelop.Components.TreeViewState.Save () [0x0000c] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components/TreeViewState.cs:59 <br>  
at MonoDevelop.VersionControl.Git.GitConfigurationDialog.FillRemotes () [0x0000e] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitConfigurationDialog.cs:147 <br>  
at MonoDevelop.VersionControl.Git.GitConfigurationDialog.OnButtonFetchClicked (System.Object sender, System.EventArgs e) [0x00101] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitConfigurationDialog.cs:376 <br>  
at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] 
in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-10/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1023 <br>  
at MonoDevelop.Ide.DispatchService+GtkSynchronizationContext+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00016] 
in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77 |
```

> VS bug [#789301](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789301)

**Comment by jsuarezruiz** [comment]
Sent PR: https://github.com/mono/monodevelop/pull/7128

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1498 (master)

Pull Request #7128 merged by: Vsevolod Kukol
Author: mono
Commit: 5ff97e1b2c7cd0fd7e80fd8d911a87095497ef70 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2306 (release-8.0)

Pull Request #7128 merged by: Vsevolod Kukol
Author: mono
Commit: 5ff97e1b2c7cd0fd7e80fd8d911a87095497ef70 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7128 merged by: Vsevolod Kukol
Author: mono
Commit: 5ff97e1b2c7cd0fd7e80fd8d911a87095497ef70 (xamarin/monodevelop)
